src/Bicep.LangServer/Handlers/BicepBuildCommandHandler.cs

```diff
@@ -0,0 +1,125 @@
+
+            DocumentUri documentUri = DocumentUri.FromFileSystemPath(bicepFilePath);
+            string buildOutput = GenerateCompiledFileAndReturnBuildOutputMessage(bicepFilePath, documentUri);
+
+            return Task.FromResult(buildOutput);
+        }
+
+        private string GenerateCompiledFileAndReturnBuildOutputMessage(string bicepFilePath, DocumentUri documentUri)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            string? compiledFilePath = GetCompiledFilePath(bicepFilePath, sb, out string bicepFile, out string compiledFile);
+
+            if (string.IsNullOrWhiteSpace(compiledFilePath))
+            {
+                throw new ArgumentException($"Invalid input file");
+            }
+
+            CompilationContext? context = CompilationManager.GetCompilation(documentUri);
+
+            if (context is null)
+            {
+                throw new InvalidOperationException($"Unable to get compilation context");
+            }
+
+            SemanticModel semanticModel = context.Compilation.GetEntrypointSemanticModel();
+            KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile = context.Compilation.GetAllDiagnosticsByBicepFile()
+                .FirstOrDefault(x => x.Key.FileUri == documentUri.ToUri());
+
+            if (diagnosticsByFile.Value.Any())
+            {
+                AppendDiagnosticsMessage(diagnosticsByFile, bicepFile, sb);
+                return sb.ToString();
+            }
+
+            using (FileStream fileStream = new FileStream(compiledFilePath, FileMode.Create, FileAccess.ReadWrite, FileShare.ReadWrite))
+            {
+                TemplateEmitter emitter = new TemplateEmitter(semanticModel, ThisAssembly.AssemblyFileVersion);
+                EmitResult result = emitter.Emit(fileStream);
+
+                sb.AppendLine("Build succeeded. Created transpiled ARM template: " + compiledFile);
+            }
+
+            return sb.ToString();
+        }
+
+        private void AppendDiagnosticsMessage(KeyValuePair<BicepFile, IEnumerable<IDiagnostic>> diagnosticsByFile, string bicepFile, StringBuilder sb)
+        {
+            sb.AppendLine("Build failed. Please fix below errors in " + bicepFile + " :");
+
+            IEnumerable<IDiagnostic> diagnostics = diagnosticsByFile.Value;
+            IReadOnlyList<int> lineStarts = diagnosticsByFile.Key.LineStarts;
+
+            for (int i = 0; i < diagnostics.Count(); i++)
+            {
+                IDiagnostic diagnostic = diagnostics.ElementAt(i);
+                var (startLine, startChar) = TextCoordinateConverter.GetPosition(lineStarts, diagnostic.Span.Position);
+
+                sb.AppendLine((i+1) + ". " + diagnostic.Message + " bicep(" + diagnostic.Code + "). " + "[" + (startLine + 1) + ", " + (startChar + 1) + "]");
+            }
+        }
+
+        private string? GetCompiledFilePath(string bicepFilePath, StringBuilder sb, out string bicepFile, out string compiledFile)
```

Comment:
>GetCompiledFilePath [](http://example.com/codeflow?start=24&length=19)

We have a `PathHelper` class that implements similar logic for the CLI. Can we reuse that?

Reply:
Updated as part of this commit:
https://github.com/Azure/bicep/commit/daa24e89c632c399b92f0de41c98fcea77b930eb